runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/stableinput/BufferingDoFnRunner.java

```diff
@@ -85,6 +86,8 @@ public static <InputT, OutputT> BufferingDoFnRunner<InputT, OutputT> create(
   int currentStateIndex;
   /** The current handler used for buffering. */
   private BufferingElementsHandler currentBufferingElementsHandler;
+  /** Minimum timestamp of all buffered elements. */
+  private volatile long currentOutputWatermarkHold;
```

Comment:
Should we make this optional so that we can differentiate in the case where the are no elements to prevent the watermark from advancing to Long.MAX_VALUE?

Reply:
Your right, my confusion was about the purpose here. It would make more sense to call this `minOutputWatermarkHoldSeen`

Reply:
changed the name to `minBufferedElementTimestamp`